### PR TITLE
Subcool reheat dx coil follow up

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Address comments for https://github.com/NREL/EnergyPlus/pull/7707#pullrequestreview-379471544.

•	Moved 3 global variables from CoilCoolingDX to DataHVACGlobals  

    extern int const coilNormalMode;        // Normal operation mode
    extern int const coilEnhancedMode;      // Enhanced operation mode
    extern int const coilSubcoolReheatMode; // SubcoolReheat operation mode

•	Removed a new coil type of CoilDX_SubcoolReheat in DataHVACGlobals. Instead, added a new variable of SubcoolReheatFlag in the Coil:Cooling:DX 
•	Removed #include <EnergyPlus/Coils/CoilCoolingDX.hh> in the Coils/CoilCoolingDXCurveFitPerformance to avoid a circular reference

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix - some minor doc defects, everything else is refactoring.
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

### CODE REVIEW

**Comment by lgu1234** [comment]
@Myoldmopar I addressed some your comments. Other comments may be addressed later as 
as the coil work is moved forward. I also addressed Rich's comments. You may take a quick look to see if is worth checking in the current work or not.

**Comment by nrel-bot-3** [comment]
@lgu1234 @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@lgu1234 @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@lgu1234 @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@lgu1234 @lgentile it has been 28 days since this pull request was last updated.

**Comment by mjwitte** [comment]
@lgu1234 Please update this and resolve conflicts.

File: src/EnergyPlus/DataHVACGlobals.hh [hunk 1]
```diff
@@ -229,7 +229,10 @@ namespace DataHVACGlobals {
 
     extern int const CoilVRF_FluidTCtrl_Cooling;
     extern int const CoilVRF_FluidTCtrl_Heating;
-    extern int const CoilDX_SubcoolReheat;
+
+    extern int const coilNormalMode;        // Normal operation mode
+    extern int const coilEnhancedMode;      // Enhanced operation mode
+    extern int const coilSubcoolReheatMode; // SubcoolReheat operation mode
```
Comment (mjwitte) [inline src/EnergyPlus/DataHVACGlobals.hh:235]: @Myoldmopar These are moved here from CoilCoolingDX.hh. Are you ok with that? 
Would we prefer an `enum class` here?

**Review by mjwitte — COMMENTED** [review]
Doc changes look good. Unit tests are running locally now.
@Myoldmopar I think this addresses your comments from https://github.com/NREL/EnergyPlus/pull/7707#pullrequestreview-379471544.

File: src/EnergyPlus/DataHVACGlobals.hh (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/DataHVACGlobals.hh:235]: An enum class is always preferred when it is possible, but it doesn't _necessarily_ be fixed as a part of this.

**Comment by Myoldmopar** [comment]
I'm going to resolve the conflicts with develop locally and testing, and looking at the code at the same time.  Anything else remaining before final review here?  (@mjwitte @lgu1234)

**Comment by mjwitte** [comment]
Nothing more from me.

**Comment by lgu1234** [comment]
@Myoldmopar No.

**Comment by Myoldmopar** [comment]
Unit tests run well locally after resolving conflicts.  I've pushed it up to get a fresh set of CI results, and I'll look over the code and try to run a few other tests locally.

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -8533,7 +8535,7 @@ namespace UnitarySystems {
                                                       CompressorONFlag);
                         PartLoadRatio = CoolPLR;
                     } else if (DataGlobals::DoCoilDirectSolutions && CoolingLoad &&
-                               this->m_CoolingCoilSubType_Num == DataHVACGlobals::CoilDX_SubcoolReheat && this->m_NumOfSpeedCooling == 1) {
+                               this->m_CoolingCoilSubType_Num == DataHVACGlobals::CoilDX_Cooling && this->m_NumOfSpeedCooling == 1) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/UnitarySystem.cc]: I had to make this change manually after pulling in develop.  I think I have it wrong though.  This should also be checking the new `SubcoolReheatFlag`, right?

**Review by Myoldmopar — COMMENTED** [review]
Other than the one thing I noted, I think this is good to go.  @lgu1234 can you confirm the note I made here?  If so, I can make the final change.  But maybe it is good to go and I just misunderstood.

File: src/EnergyPlus/UnitarySystem.cc (hunk 2 above)
Comment (lgu1234) [reply src/EnergyPlus/UnitarySystem.cc]: @Myoldmopar Yes. I am going to work on it on my local computer and let you know what happens.

**Comment by lgu1234** [comment]
@Myoldmopar I just uploaded my changes a while ago. Please take a look.

**Comment by Myoldmopar** [comment]
Nice!  That was my only concern, I will merge this in.  Thanks @lgu1234.

**Comment by Myoldmopar** [comment]
Given that #8240 might have a little more work to do, I'm going to go ahead and merge this one in.

**Pull request merged**